app/main.py

```diff
@@ -8,49 +12,49 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f"Distance(km={self.km})"
 
-    def __eq__(self, other: float) -> bool:
+    def __eq__(self, other: Union[float, int, Distance]) -> bool:
```

Comment:
let's use | operator instead of Union class (it's the same) and fix it everywher